tasks/upload_finisher.py

```diff
@@ -568,15 +602,16 @@ def merge_report(cumulative_report: Report, obj):
                 clear_carryforward_sessions(
                     cumulative_report, incremental_report, flags, UserYaml(commit_yaml)
                 )
-            # ReportService.update_upload_with_processing_result should use this result
-            # to update the state of Upload. Once the experiment is finished, Upload.state should
-            # be set to: parallel_processed (instead of processed)
 
             cumulative_report.merge(incremental_report)
 
-            # once the experiment is finished, we should be modifying the Upload here
-            # moving it's state from: parallel_processed -> processed
-
+            if parallel_processing is ParallelProcessing.PARALLEL:
+                # When we are fully parallel, we need to update the `Upload` in the database
+                # with the final session_id (aka `order_number`).
```

Comment:
is this taking the place of the `PARALLEL_PROCESSED` upload state daniel had?

Reply:
not quite. I will discuss these various states a bit more and figure out a good way to go there.

Reply:
But good that you called this out, I found another bug related to the new code from #745 not being ported to this PR yet, which I now did.